src/wavelet_matrix.rs

```diff
@@ -0,0 +1,559 @@
+use anyhow::{anyhow, Result};
+use std::ops::Range;
+
+use crate::{broadword, BitVector, RsBitVector};
+
+/// Time and space efficient index data structures for sequence.
+///
+/// [`WaveletMatrix`] implements stores a sequence of integers and provides myriad operations on sequece.
+/// When a sequence stores $`n`$ integers from $`[0, u-1]`$,
+/// most of the operations run in $`O(log u)`$ , using  $`n log u + o (n log u) + O(log u log n)`$ bits.
+///
+/// This is a yet another Rust port of [hillbig's waveletMatrix](https://github.com/hillbig/waveletTree/blob/master/waveletMatrix.go).
+///
+/// # Example
```

Comment:
```suggestion
/// # Examples
```

For consistency with other files.